stan/math/prim/mat/fun/MatrixExponential.h

```diff
@@ -0,0 +1,211 @@
+    const RealScalar b[] = {17643225600.L, 8821612800.L, 2075673600.L, 302702400.L, 30270240.L,  
+    2162160.L, 110880.L, 3960.L, 90.L, 1.L};
+    const MatrixType A2 = A * A;
+    const MatrixType A4 = A2 * A2;
+    const MatrixType A6 = A4 * A2;
+    const MatrixType A8 = A6 * A2;
+    const MatrixType tmp = b[9] * A8 + b[7] * A6 + b[5] * A4 + b[3] * A2
+    + b[1] * MatrixType::Identity(A.rows(), A.cols());
+    U.noalias() = A * tmp;
+    V = b[8] * A8 + b[6] * A6 + b[4] * A4 + b[2] * A2 + b[0] * MatrixType::Identity(A.rows(), A.cols());  
+  }
+
+  /** \brief Compute the (13,13)-Pad&eacute; approximant to the exponential.
+   *
+   *  After exit, \f$ (V+U)(V-U)^{-1} \f$ is the Pad&eacute;
+   *  approximant of \f$ \exp(A) \f$ around \f$ A = 0 \f$.
+   */
+  template <typename MatrixType>
+  void matrix_exp_pade13(const MatrixType &A, MatrixType &U, MatrixType &V)  
+  {  
+    using Eigen::internal::traits; 
+    typedef typename Eigen::NumTraits<typename traits<MatrixType>::Scalar>::Real RealScalar;  
+    const RealScalar b[] = {64764752532480000.L, 32382376266240000.L, 7771770303897600.L,  
+    1187353796428800.L, 129060195264000.L, 10559470521600.L, 670442572800.L,  
+    33522128640.L, 1323241920.L, 40840800.L, 960960.L, 16380.L, 182.L, 1.L};  
+    const MatrixType A2 = A * A;
+    const MatrixType A4 = A2 * A2;
+    const MatrixType A6 = A4 * A2;
+    V = b[13] * A6 + b[11] * A4 + b[9] * A2; // used for temporary storage  
+    MatrixType tmp = A6 * V;
+    tmp += b[7] * A6 + b[5] * A4 + b[3] * A2 + b[1] * MatrixType::Identity(A.rows(), A.cols());  
+    U.noalias() = A * tmp;
+    tmp = b[12] * A6 + b[10] * A4 + b[8] * A2;
+    V.noalias() = A6 * tmp;
+    V += b[6] * A6 + b[4] * A4 + b[2] * A2 + b[0] * MatrixType::Identity(A.rows(), A.cols());  
+  }
+
+  template <typename MatrixType, typename RealScalar = typename Eigen:: NumTraits<typename Eigen::internal::traits<MatrixType>::Scalar>::Real>  
+  struct matrix_exp_computeUV
+  {  
+    /** \brief Compute Pad&eacute; approximant to the exponential.
+     *
+     * Computes \c U, \c V and \c squarings such that \f$ (V+U)(V-U)^{-1} \f$ is a Pad&eacute;
+     * approximant of \f$ \exp(2^{-\mbox{squarings}}M) \f$ around \f$ M = 0 \f$, where \f$ M \f$
+     * denotes the matrix \c arg. The degree of the Pad&eacute; approximant and the value of squarings
+     * are chosen such that the approximation error is no more than the round-off error.
+     *
+     * <p> Edit for Stan: template ComputeUV::run so that it may used on
+     * autodiff variables (var and fvar). This required adding the scalar_type
+     * argument, which tells the function the type of elements used in the
+     * matrix.
+     */
+    static void run(const MatrixType& arg, MatrixType& U, MatrixType& V, int& squarings);  
+  };
+
+  template <typename MatrixType>
+  struct matrix_exp_computeUV<MatrixType>
+  {  
+  	template <typename T>  
```

Comment:
my fault. I added this template a while ago, and forgot to change the tab to two spaces.